tests/unit/test_codeact_agent.py

```diff
@@ -3,20 +3,45 @@
 import pytest
 
 from openhands.agenthub.codeact_agent.codeact_agent import CodeActAgent
+from openhands.agenthub.codeact_agent.function_calling import (
+    _BROWSER_DESCRIPTION,
+    _BROWSER_TOOL_DESCRIPTION,
+    BrowserTool,
+    CmdRunTool,
+    IPythonTool,
+    LLMBasedFileEditTool,
+    StrReplaceEditorTool,
+    WebReadTool,
+    get_tools,
+    response_to_actions,
+)
 from openhands.core.config import AgentConfig, LLMConfig
-from openhands.core.message import TextContent
+from openhands.core.exceptions import FunctionCallNotExistsError
+from openhands.core.message import ImageContent, TextContent
+from openhands.events.action import (
+    AgentFinishAction,
+    CmdRunAction,
+    MessageAction,
+)
+from openhands.events.event import EventSource, FileEditSource, FileReadSource
+from openhands.events.observation.browse import BrowserOutputObservation
 from openhands.events.observation.commands import (
     CmdOutputObservation,
     IPythonRunCellObservation,
 )
 from openhands.events.observation.delegate import AgentDelegateObservation
 from openhands.events.observation.error import ErrorObservation
+from openhands.events.observation.files import FileEditObservation, FileReadObservation
+from openhands.events.observation.reject import UserRejectObservation
+from openhands.events.tool import ToolCallMetadata
 from openhands.llm.llm import LLM
 
 
 @pytest.fixture
 def agent() -> CodeActAgent:
-    agent = CodeActAgent(llm=LLM(LLMConfig()), config=AgentConfig())
+    config = AgentConfig()
+    config.use_microagents = False  # Disable microagents for testing
+    agent = CodeActAgent(llm=LLM(LLMConfig()), config=config)
```

Comment:
```suggestion
    agent = CodeActAgent(llm=LLM(LLMConfig()), config=AgentConfig())
```

This comment doesn't convince me. I suspect this is not needed (and bad because it alters default behavior in testing). Otherwise, if there's a compelling reason, the comment could be clearer.